### ISSUE TITLE
old developer install failing storage opening

### ISSUE THREAD

**Issue by sroet** [openpathsampling/openpathsampling#888]
This is just for documentation, as I could not reproduce after applying the solution: `pip install -e .`

After I pulled in the new master, any tests that involved reopening storage (either in `a` or `r` mode, or `AnalysisStorage`) started failing with:
```python
openpathsampling/tests/test_transitions.py:70: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
openpathsampling/storage/storage.py:495: in __init__
    mode='r'
openpathsampling/storage/storage.py:64: in __init__
    fallback=fallback)
openpathsampling/netcdfplus/netcdfplus.py:307: in __init__
    self.check_version()
openpathsampling/storage/storage.py:212: in check_version
    cp = self._cmp_version(s1, s2)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

v1 = 'Unknown+gd32b329', v2 = 'Unknown'

    @staticmethod
    def _cmp_version(v1, v2):
        # we only look at x.y.z parts
        q1 = v1.split('.')[:3]
        q2 = v2.split('.')[:3]
        # q1 = v1.split('-')[0].split('.')
        # q2 = v2.split('-')[0].split('.')
        for v1, v2 in zip(q1, q2):
>           if int(v1) > int(v2):
E           ValueError: invalid literal for int() with base 10: 'Unknown+gd32b329'

openpathsampling/netcdfplus/netcdfplus.py:393: ValueError
```

This was solved by redoing `pip install -e .` and I could not figure out a way to reproduce it after that...

**Comment by dwhswenson** [comment]
This makes sense: Storage checks that you are loading using a version of OPS that is compatible with the version that wrote the file. 

Version information has always been written by setup.py. When we switched to using Autorelease in #883, we changed how that was done. The old way was to write a version.py file. The new way is to have a standard `version.py` file, and write an `_installed_version.py`. If `_installed_version.py` does not exist, the version is “Unknown”, hence can’t be parsed as a version, hence your error.

It used to be that version information was just a fixed string, which meant that developer installs would forever claim to be whatever version they were when you installed them. This was once of the reasons for doing #883. (I'm not positive, but that might have meant that a developer install that was installed at version 0.9.3, but is up-to-date with the source from 1.1, wouldn't be able to load files from 0.9.6.)

I rewrote the version parsing as part of the release PR. Normally I’d immediately merge that back into master with a version bump, but I’m waiting to see if conda-forge is going to force me to cut a bump release (forgot to include license in manifest) or of they’ll accept license-in-recipe directory (their CI technically does, but they don't like it).
